integration/spring-data/base/src/main/java/com/blazebit/persistence/spring/data/base/query/AbstractPartTreeBlazePersistenceQuery.java

```diff
@@ -372,6 +385,20 @@ private Sort getDynamicSort(Object[] values) {
             return parameters.potentiallySortsDynamically() ? new ParametersParameterAccessor(parameters, values).getSort()
                     : null;
         }
+
+        private <T> void applySpecification(CriteriaQuery<T> criteriaQuery, Object[] values) {
+            JpaParameter specificationParameter = parameters.getSpecificationParameter();
+            if (specificationParameter != null) {
+                int specificationParameterIndex = specificationParameter.getIndex();
+                Specification<T> specification = (Specification<T>) values[specificationParameterIndex];
+                Root<T> root = (Root<T>) criteriaQuery.getRoots().iterator().next();
+                CriteriaBuilder criteriaBuilder = getEntityManager().getCriteriaBuilder();
+                Predicate predicate = specification.toPredicate(root, criteriaQuery, criteriaBuilder);
+                if (predicate != null) {
```

Comment:
Either you always replace the original predicate i.e. the one created by the PartTree by also setting null or you always add the predicate by doing something like this `criteriaQuery.where(criteriaBuilder.and(criteriaQuery.getWhere(), predicate))`